index.js

```diff
@@ -24,48 +24,40 @@ SOFTWARE.
 
 'use strict';
 
-function skip(input, skipIndex) {
-    return input.filter(function(item, index) {
-        return Boolean(index > skipIndex - 1);
-    });
-}
+function between(arr, start, end) {
+    validateArguments(arr, start, end);
 
-function limit(input, limitIndex) {
-    return input.filter(function (item, index) {
-        return Boolean(index <= limitIndex);
-    });
-}
+    const skipped = skip(arr, start);
 
-function throwTypeError(message) {
-    throw new TypeError(message);
+    return limit(skipped, end - start);
 }
 
+
 function validateArguments(arr, start, end) {
-     if (!Array.isArray(arr)) {
-        throwTypeError(`Expected an Array, got ${typeof arr}`);
+    if (!Array.isArray(arr)) {
+        new TypeError(`Expected an Array, got ${typeof arr}`);
```

Comment:
typeof array is `'object'`. Consider printing the toString value (`[object Object]` VS `[object Array]`)
```suggestion
        new TypeError(`Expected an Array, got ${arr}`);
```